modules/mapper-extras/src/test/resources/rest-api-spec/test/search-as-you-type/20_highlighting.yml

```diff
@@ -0,0 +1,74 @@
+setup:
+  - skip:
+      version: " - 7.0.99"
+      reason: "search-as-you-type field type was introduced in 7.1.0"
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          settings:
+            number_of_replicas: 0
+          mappings:
+            properties:
+              a_field:
+                type: search_as_you_type
+                analyzer: simple
+                max_shingle_size: 4
+              text_field:
+                type: text
+                analyzer: simple
+
+  - do:
+      index:
+        index: test
+        type: _doc
+        id: 1
+        body:
+          a_field: "quick red fox lazy brown dog"
+          text_field: "quick red fox lazy brown dog"
+
+  - do:
+      indices.refresh: {}
+
+---
+"phrase query unified with full term":
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          query:
+            match_phrase:
+              a_field: "red"
+          highlight:
+            fields:
+              a_field:
+                type: unified
+
+  - match: { hits.total: 1 }
+  - match: { hits.hits.0._source.a_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0._source.text_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0.highlight.a_field.0: "quick <em>red</em> fox lazy brown dog" }
+
+---
+"boolean prefix query unified with full term":
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          query:
+            boolean_prefix:
+              a_field: "red fo"
+          highlight:
+            fields:
+              a_field:
+                type: unified
+
+  - match: { hits.total: 1 }
+  - match: { hits.hits.0._source.a_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0._source.text_field: "quick red fox lazy brown dog" }
+  - match: { hits.hits.0.highlight.a_field.0: "quick <em>red</em> fox lazy brown dog" }
```

Comment:
Can you add a test with a `multi_match` query that targets all shingle fields and sets the type to `boolean_prefix` ?